pkg/image/registry/image/etcd/etcd.go

```diff
@@ -59,8 +59,27 @@ func NewREST(optsGetter restoptions.Getter) (*REST, error) {
 	}
 
 	if err := restoptions.ApplyOptions(optsGetter, store, prefix); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	internalStore := *store
+	internalStore.UpdateStrategy = image.InternalStrategy
+	internalStore.CreateStrategy = nil
+
+	return &REST{store}, &InternalREST{store: &internalStore}, nil
+}
+
+// InternalREST implements a RESTStorage for image resources internally modifiable.
```

Comment:
Why can't I do all that using the current `REST` being returned?  Seems like it has an update function, right?